### PR TITLE
[AC-636] Add a Splash Screen

### PR DESCRIPTION
**Pull request by ribhavsharma**
## Description of what I changed
Added a Splash Screen to the app, with an animation.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-636
GCI Task: https://codein.withgoogle.com/dashboard/task-instances/6153655376412672/

## Screenshot
ezgif-3-d720da0edfa3

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import android.view.animation.AnimationSet;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import org.openmrs.mobile.R;
+import org.openmrs.mobile.activities.login.LoginActivity;
+
+
```
Comment (suthagar23) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Remove this line, or better to add class level doc here.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+public class SplashActivity extends AppCompatActivity {
+
+    Animation move, fadeIn;
+    ImageView logo;
+    @Override
```
Comment (suthagar23) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Need a new line between variables and methods

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        new Thread(){
+            @Override
+            public void run() {
+                super.run();
+                try {
+                    Thread.sleep(4500);
+                    Intent intent = new Intent(SplashActivity.this, LoginActivity.class);
+                    startActivity(intent);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```
Comment (suthagar23) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: What is the use of this while a user running the mobile application?

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                }
+            }
+        }.start();
+
+    }
+
```
Comment (suthagar23) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Remove unwanted new lines in your changes

File: openmrs-client/src/main/res/layout/activity_splash.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_height="match_parent"
+    tools:context=".activities.SplashActivity">
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:src="@drawable/ic_openmrs"
+        android:layout_centerInParent="true"
+        android:id="@+id/logo"/>
+
+</RelativeLayout>
```
Comment (suthagar23) [inline openmrs-client/src/main/res/layout/activity_splash.xml]: New line, check for other files also

File: openmrs-client/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -1,16 +1,4 @@
...
-  ~ compliance with the License. You may obtain a copy of the License at
-  ~ http://license.openmrs.org
-  ~
-  ~ Software distributed under the License is distributed on an "AS IS"
-  ~ basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing rights and limitations
-  ~ under the License.
-  ~
-  ~ Copyright (C) OpenMRS, LLC.  All Rights Reserved.
--->
```
Comment (f4ww4z) [inline openmrs-client/src/main/AndroidManifest.xml]: Don't remove this copyright.

File: openmrs-client/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -41,8 +38,7 @@
             android:value="@integer/aadbversion" />
         <meta-data
             android:name="AA_MODELS"
-            android:value="org.openmrs.mobile.models.Link, org.openmrs.mobile.models.FormResource, org.openmrs.mobile.models.EncounterType,
-             org.openmrs.mobile.models.Encountercreate, org.openmrs.mobile.models.Obscreate" />
+            android:value="org.openmrs.mobile.models.Link, org.openmrs.mobile.models.FormResource, org.openmrs.mobile.models.EncounterType,              org.openmrs.mobile.models.Encountercreate, org.openmrs.mobile.models.Obscreate" />
```
Comment (f4ww4z) [inline openmrs-client/src/main/AndroidManifest.xml]: This does not need to be changed. It was properly formatted before.

File: openmrs-client/build.gradle [hunk 8]
```diff
@@ -114,6 +114,8 @@ dependencies {
     def room_version = "2.1.0-alpha04"
 
     implementation 'com.google.code.gson:gson:2.8.5'
+    implementation 'com.android.support:appcompat-v7:28.X.+'
```
Comment (f4ww4z) [inline openmrs-client/build.gradle]: Use the **AndroidX** library instead. https://developer.android.com/jetpack/androidx/releases/appcompat

File: openmrs-client/build.gradle [hunk 9]
```diff
@@ -114,6 +114,8 @@ dependencies {
...
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (f4ww4z) [inline openmrs-client/build.gradle]: Use the AndroidX counterpart instead. See [line 140](https://github.com/openmrs/openmrs-contrib-android-client/pull/638/files#diff-cc5437861bff23b3bb39d227f1f32eaeR140). No need to add new dependencies.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        fadeIn.setDuration(1000);
+        set.addAnimation(fadeIn);
+        set.addAnimation(move);
+        logo.startAnimation(set);
+        new Thread(){
+            @Override
+            public void run() {
+                super.run();
+                try {
+                    Thread.sleep(4500);
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Make it `3500` instead.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+import android.widget.ImageView;
+import androidx.appcompat.app.AppCompatActivity;
+import org.openmrs.mobile.R;
+import org.openmrs.mobile.activities.login.LoginActivity;
+
+
+public class SplashActivity extends AppCompatActivity {
+
+    Animation move, fadeIn;
+    ImageView logo;
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Turn these variables to local variables if possible.

File: openmrs-client/src/main/res/anim/logo_anim.xml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/anim/logo_anim.xml]: Add a copyright here. Also rename the file to `splash_screen_logo_anim`

File: openmrs-client/src/main/res/layout/activity_splash.xml [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/layout/activity_splash.xml]: Add a copyright notice here.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java (hunk 1 above)
Comment (suthagar23) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Did you address this comment?

File: openmrs-client/src/main/res/layout/activity_splash.xml (hunk 5 above)
Comment (suthagar23) [reply openmrs-client/src/main/res/layout/activity_splash.xml]: Did you address this comment?

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        fadeIn.setDuration(1000);
+        set.addAnimation(fadeIn);
+        set.addAnimation(move);
+        logo.startAnimation(set);
+        new Thread(){
+            @Override
+            public void run() {
+                super.run();
+                try {
+                    Thread.sleep(3500);
```
Comment (suthagar23) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: move this value to a constant

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java (hunk 1 above)
Comment (ribhavsharma) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Hey @suthagar23. I didn't quite get you. Could you explain it a bit more?

File: build.gradle [hunk 15]
```diff
@@ -8,7 +8,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.2'
+        classpath 'com.android.tools.build:gradle:3.2.0'
```
Comment (f4ww4z) [inline build.gradle]: We don't need to downgrade this.

File: openmrs-client/build.gradle [hunk 16]
```diff
@@ -146,6 +148,7 @@ dependencies {
     implementation 'com.michaelpardo:activeandroid:3.1.0-SNAPSHOT'
     implementation 'com.squareup.okhttp3:logging-interceptor:3.8.0'
     implementation 'androidx.multidex:multidex:2.0.0'
+    implementation "com.android.support:support-v4:28.0.+"
```
Comment (f4ww4z) [inline openmrs-client/build.gradle]: Which part of the support library do you need? Can you use `AndroidX` for this instead? If not, decrease the bundle size by specifying the exact components you need [as described here](https://developer.android.com/topic/libraries/support-library/packages#v4). Also, use fixed versions e.g. `28.0.0`

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+import android.view.animation.Animation;
+import android.view.animation.AnimationSet;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+import androidx.appcompat.app.AppCompatActivity;
+import org.openmrs.mobile.R;
+import org.openmrs.mobile.activities.login.LoginActivity;
+
+public class SplashActivity extends AppCompatActivity {
+    int SPLASH_TIMER = 3500;
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/SplashActivity.java]: Make this a constant declared at [ApplicationConstants](https://github.com/openmrs/openmrs-contrib-android-client/blob/master/openmrs-client/src/main/java/org/openmrs/mobile/utilities/ApplicationConstants.java).

File: openmrs-client/src/main/res/anim/splash_screen_logo_anim.xml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/anim/splash_screen_logo_anim.xml]: Add a copyright notice here.

**Pull request merged**